lobster/io.py

```diff
@@ -110,14 +110,24 @@ def lobster_read(mh, filename, level, items, source_info=None):
         else:
             item = Activity.from_json(level, raw, data["version"])
 
-        if item.tag.key() in items:
-            mh.error(item.location,
-                     "duplicate definition of %s, "
-                     "previously defined at %s" %
-                     (item.tag.key(),
-                      items[item.tag.key()].location.to_string()))
+        filter_conditions = []
 
         if source_info is not None:
             item.perform_source_checks(source_info)
-
-        items[item.tag.key()] = item
+    
+            # evaluate source_info filters
+            for f, v in source_info['filters']:
+                if f == 'prefix':
+                    filter_conditions.append(item.tag.tag.startswith(v))
+                if f == 'kind':
```

Comment:
I know this has been done all over the place, but do you see a chance to introduce constants, such that the string `kind` does not need to be created twice? It is like a hard-coded number (although it's not a number), and it is needed in `parser.py`as well. As I said, this coding style has been done very often so far, but I would like to improve our style and get rid of magic strings.

Reply:
For introducing string constants I don't see a straight forward approach:
- An Item is parsed in the from_json() method after reading the LOBSTER / json file. 
- The config parser code is independent of the Item and classes derived from Item. However the config syntax contains attributes from Item. This already existed with the 'with valid_status'. With adding 'with prefix' and 'with kind' this got worse.

To improve the filter implementation I will move the code you cite above from io.py to items.py, Item.perform_source_checks() and return a boolean. Then the definition of filter conditions are part of the Item class.

Do you agree?

Reply:
I see your point. Don't worry, keep it as it is. Otherwise we would mix a feature implementation with a refactoring, which should be separate PRs with separate reviews, and especially with separate commits in the history.